hexrd/ui/image_stack_dialog.py

```diff
@@ -437,3 +439,20 @@ def show(self):
     def close_widget(self):
         if self.ui.isFloating():
             self.ui.close()
+
+    def detectors_changed(self):
+        self.detectors = HexrdConfig().detector_names
+        self.detector = self.detectors[0]
+        # update the state
+        self.state['dets'] = self.detectors
+        for det in self.detectors:
+            self.state[det] = {
+                'directory': '',
+                'files': '',
+                'search': '',
+                'file_count': 0,
+            }
```

Comment:
I see this is copied from `setup_state()`. Maybe we should put it in a function or property to reduce duplicated code? This would be especially helpful in case we decided to change the default detector settings in the future.

Reply:
Yes, sounds good. Moved into its own function `setup_detectors_state`